File: paiza/ruby/C050.rb
# 自分の得意な言語で
# Let's チャレンジ！！

ipt = gets.chomp.split(' ')
price = ipt[0].to_i
a_max = ipt[1].to_i
b_max = ipt[2].to_i

while true do
 
    if (price + 10) > a_max
        printf('B %d', price)
        break
    else
        price += 10
    end
    
    if (price + 1000) > b_max
        printf('A %d', price)
        break
    else
        price += 1000
    end
end
    
    

File: paiza/javascript/C050.js
process.stdin.resume();
process.stdin.setEncoding('utf8');

var ipt = []; // 標準入力から与えられた各行を保存する変数：input_lines
var reader = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});
reader.on('line', (line) => {
  ipt.push(line);
});
reader.on('close', () => {
    const input = ipt[0].split(' ');
    let price = parseInt(input[0]);
    const max_A = parseInt(input[1]);
    const max_B = parseInt(input[2]);
    const step_A = 10;
    const step_B = 1000;
    let next_price = 0;
    
    while (true) {
        next_price = price + step_A;
        if (next_price > max_A) {
            console.log("B " + price);
            break;
        }
        
        price = next_price;
        next_price = price + step_B;
        if (next_price > max_B) {
            console.log("A " + price);
            break;
        }
        
        price = next_price;
    }
});